common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -883,6 +918,25 @@ def answer_available(self):
 
         return False
 
+    def correctness_available(self):
+        """
+        Is the user allowed to see whether she's answered correctly?
+
+        Limits access to the correct/incorrect flags, messages, and problem score.
+        """
+        if self.show_correctness == SHOW_CORRECTNESS.NEVER:
+            return False
+        elif self.runtime.user_is_staff:
+            # This is after the 'never' check because admins can see correctness
```

Comment:
@cahrens @pomegranited I didn't know that you could set a coursewide default for `SHOWANSWER` in the course advanced settings (though I checked Studio and lo and behold it's true!), and we don't document that anywhere that I can find. 

So that I'm clear on this: The goal here is to hide results for individual problems from learners (and from course teams viewing the course as learners) until after the due date, but to make the results visible for staff at all times. To do this, the course team has to set **Show Answer** in the course advanced settings to `closed` (?), then set **Show Results** for the subsection to **Show results when subsection is past due**, and then set **Show Answer** in the settings for each individual problem to **Never** or **Past Due**? I feel like I'm missing something; I'm not sure why the individual problem setting is necessary.